### DISCUSSION TITLE
Unable to create an object of type 'DataContext'

### DISCUSSION THREAD

**Discussion by BrunoBafilli** [balta-io/forum#267] [Tira Dúvidas] [answered]
Estou tendo o seguinte erro no memento da criação de uma Migrations usando EF Core, ao digitar o comando `dotnet ef migrations add initial
`
**Unable to create an object of type 'DataContext'. For the different patterns supported at design time, see https://go.microsoft.com/fwlink/?linkid=851728**

Alguem sabe o motivo? e qual é a solução?

**Comment by andrebaltieri** [comment]
Boa tarde, @BrunoBafilli, como vai?

Este erro ocorre por que a migration não conseguiu instanciar seu DbContext.
Os motivos podem ser:
- Ele não tem um construtor sem parâmetros (Parameterless)
- Sua migration está em um projeto diferente do seu DbContext

Como está seu projeto? Consegue anexar um print?

**Comment by BrunoBafilli** [comment, marked as answer]
Boa tarde @andrebaltieri 

O problema era esse - Ele não tem um construtor sem parâmetros (Parameterless)

Muito obrigado!

**Reply by andrebaltieri** [reply]
Perfeito!
